### PR TITLE
Styled Header for new display package page

### PR DESCRIPTION
**Pull request by sophiamfavro**
# Solution 
This PR is all about styling the header of the new display package page. I changed the sizing, spacing, and colors of the header elements to match the redesign mock ups. I also added a "prefix reserved" label that links to the ID prefix reservation docs. The goal of this PR was do all the necessary changes in the header except for the installation instructions, so I have left them untouched and will do those changes in a later PR. Here is what the final product looks like 
image

# Testing 
I tested all the links to make sure that they worked with the feature flag on and off, checked the console to make sure that there were no errors in the .js file, and resized the page to make sure the styling stayed the same. 
This addresses issue #8601

### CODE REVIEW

File: src/NuGetGallery/Views/Packages/DisplayPackageV2.cshtml [hunk 1]
```diff
@@ -243,261 +243,269 @@
...
-            </h3>
-        </aside>
-        <article class="col-sm-11 package-details-main">
+        <article class="col-sm-12 package-details-main">
             <div class="row">
-                <div class="package-title">
-                    <h1>
-                        @Html.BreakWord(Model.Id)
-                        <small class="text-nowrap">@Model.Version</small>
+                <div class="header">
```
Comment (loic-sharma) [inline src/NuGetGallery/Views/Packages/DisplayPackageV2.cshtml:248]: I would avoid "common" names for classes and ids. Someone could style another `header` class somewhere else on the website and mess up this header too by accident. Consider a name like `package-header`

File: src/NuGetGallery/Views/Packages/DisplayPackageV2.cshtml [hunk 2]
```diff
@@ -243,261 +243,269 @@
...
+                    <div class="package-title">
+                        <h1>
+                            <span class="pull-left">
+                                <img class="package-icon img-responsive" aria-hidden="true" alt=""
+                                     src="@(PackageHelper.ShouldRenderUrl(Model.IconUrl) ? Model.IconUrl : Url.Absolute("~/Content/gallery/img/default-package-icon.svg"))"
+                                     @ViewHelpers.ImageFallback(Url.Absolute("~/Content/gallery/img/default-package-icon-256x256.png")) />
+                            </span>
+
+                            <span id="title">
```
Comment (loic-sharma) [inline src/NuGetGallery/Views/Packages/DisplayPackageV2.cshtml:257]: We tend to use classes more often than IDs (for an ID you need to make sure it is unique on the page, there can only be one element with a specific ID, whereas it's OK if there are multiple elements with the same class). Please swap these to classes

File: src/NuGetGallery/Views/Packages/DisplayPackageV2.cshtml (hunk 1 above)
Comment (sophiamfavro) [reply src/NuGetGallery/Views/Packages/DisplayPackageV2.cshtml]: done 👍

File: src/NuGetGallery/Views/Packages/DisplayPackageV2.cshtml (hunk 2 above)
Comment (sophiamfavro) [reply src/NuGetGallery/Views/Packages/DisplayPackageV2.cshtml]: done 👍

File: src/Bootstrap/less/theme/page-display-package-v2.less [hunk 3]
```diff
@@ -11,15 +11,60 @@
...
+      .version-title {
+        font-size: 16px;
+        font-weight: 400;
+        color: #666666;
+        margin-right: 14px;
+      }
+
+      .package-icon {
+        height: 40px;
+        width: 40px;
```
Comment (loic-sharma) [inline src/Bootstrap/less/theme/page-display-package-v2.less:39]: Is this right? It looks like the icon is 32px by 32px on the mockups:

image

File: src/Bootstrap/less/theme/page-display-package-v2.less [hunk 4]
```diff
@@ -11,15 +11,60 @@
...
+        border-radius: 100px;
+
+        .reserved-indicator {
+          width: 18px;
+          margin-right: 5px;
+          margin-left: 2px;
+        }
+
+        .prefix-reserve-label {
+          font-size: 16px;
```
Comment (loic-sharma) [inline src/Bootstrap/less/theme/page-display-package-v2.less:56]: It looks like the font size should be 14px:

image

File: src/Bootstrap/less/theme/page-display-package-v2.less [hunk 5]
```diff
@@ -11,15 +11,60 @@
...
+        align-items: center;
+        border-radius: 100px;
+
+        .reserved-indicator {
+          width: 18px;
+          margin-right: 5px;
+          margin-left: 2px;
+        }
+
+        .prefix-reserve-label {
```
Comment (loic-sharma) [inline src/Bootstrap/less/theme/page-display-package-v2.less:55]: In the mockup, the version and prefix reserved labels are vertically aligned (notice how the centers are aligned):

image

However right now it looks like the bottom of the version and prefix reserved labels are aligned:

image

File: src/Bootstrap/less/theme/page-display-package-v2.less (hunk 4 above)
Comment (sophiamfavro) [reply src/Bootstrap/less/theme/page-display-package-v2.less:56]: done 👍 I was going off of the other mockup so thank you for catching this!

File: src/Bootstrap/less/theme/page-display-package-v2.less (hunk 3 above)
Comment (sophiamfavro) [reply src/Bootstrap/less/theme/page-display-package-v2.less:39]: done 👍

File: src/Bootstrap/less/theme/page-display-package-v2.less (hunk 5 above)
Comment (sophiamfavro) [reply src/Bootstrap/less/theme/page-display-package-v2.less:57]: done 👍

File: src/Bootstrap/less/theme/page-display-package-v2.less (hunk 4 above)
Comment (loic-sharma) [reply src/Bootstrap/less/theme/page-display-package-v2.less]: Oh the mockups are inconsistent? Feel free to go with whichever one you prefer then! 😅

File: src/Bootstrap/less/theme/page-display-package-v2.less [hunk 6]
```diff
@@ -11,15 +11,63 @@
...
+        font-size: 32px;
+        margin-top: 40px;
+        font-weight: 400;
+        margin-right: 18px;
+      }
+
+      .version-title {
+        font-size: 16px;
+        font-weight: 400;
+        color: #666666;
```
Comment (loic-sharma) [inline src/Bootstrap/less/theme/page-display-package-v2.less:33]: I believe you can use the `@gray-light` variable here:

```suggestion
        color: @gray-light;
```

See:
https://github.com/NuGet/NuGetGallery/blob/db62de43e24b704e9ab16d8abc39bdfa3e73260b/src/Bootstrap/less/variables.less#L16

File: src/NuGetGallery/Views/Packages/DisplayPackageV2.cshtml [hunk 7]
```diff
@@ -243,261 +243,269 @@
...
     <div class="row">
-        <aside aria-label="Package icon" class="col-sm-1">
-            <h3>
-                <img class="package-icon img-responsive" aria-hidden="true" alt=""
-                     src="@(PackageHelper.ShouldRenderUrl(Model.IconUrl) ? Model.IconUrl : Url.Absolute("~/Content/gallery/img/default-package-icon.svg"))"
-                     @ViewHelpers.ImageFallback(Url.Absolute("~/Content/gallery/img/default-package-icon-256x256.png")) />
-            </h3>
-        </aside>
-        <article class="col-sm-11 package-details-main">
+        <article class="col-sm-12 package-details-main">
```
Comment (loic-sharma) [inline src/NuGetGallery/Views/Packages/DisplayPackageV2.cshtml:246]: Can you simplify the row with a single column into just a single element? Something like:

```diff
- <div class="row">
-     <article class="col-sm-12 package-details-main">
+ <div class="package-details-main">
...
-     </article>
</div>
```

File: src/NuGetGallery/Views/Packages/DisplayPackageV2.cshtml [hunk 8]
```diff
@@ -243,261 +243,269 @@
...
+                            {
+                            <span class="prefix-reserve-title">
+                                <img class="reserved-indicator"
+                                     src="~/Content/gallery/img/reserved-indicator.svg"
+                                     @ViewHelpers.ImageFallback(Url.Absolute("~/Content/gallery/img/reserved-indicator-25x25.png"))
+                                     title="@Strings.ReservedNamespace_ReservedIndicatorTooltip" />
+                                <a href="https://docs.microsoft.com/en-us/nuget/nuget-org/id-prefix-reservation" class="prefix-reserve-label">
+                                    Prefix Reserved
+                                </a>
+                            </span>
```
Comment (loic-sharma) [inline src/NuGetGallery/Views/Packages/DisplayPackageV2.cshtml:273]: Please add a tab to this content

Comment (sophiamfavro) [reply src/NuGetGallery/Views/Packages/DisplayPackageV2.cshtml:273]: done 👍

File: src/Bootstrap/less/theme/page-display-package-v2.less (hunk 6 above)
Comment (sophiamfavro) [reply src/Bootstrap/less/theme/page-display-package-v2.less:33]: done 👍

File: src/NuGetGallery/Views/Packages/DisplayPackageV2.cshtml (hunk 7 above)
Comment (sophiamfavro) [reply src/NuGetGallery/Views/Packages/DisplayPackageV2.cshtml:246]: Will this break the grid though?

Comment (sophiamfavro) [reply src/NuGetGallery/Views/Packages/DisplayPackageV2.cshtml]: done 👍

**Review by loic-sharma — APPROVED** [review]

File: src/Bootstrap/less/theme/page-display-package-v2.less [hunk 9]
```diff
@@ -11,15 +11,63 @@
...
+
+      .package-icon {
+        height: 32px;
+        width: 32px;
+        margin-top: 8px;
+        margin-right: 12px;
+      }
+
+      .prefix-reserve-title {
+        background-color: #F3F2F1;
```
Comment (zhhyu) [inline src/Bootstrap/less/theme/page-display-package-v2.less:46]: nit: will it be better if we also use some shared color variable here? (For example: "color: @gray-light;")

File: src/Bootstrap/less/theme/page-display-package-v2.less [hunk 10]
```diff
@@ -11,15 +11,63 @@
...
+        .reserved-indicator {
+          width: 18px;
+          margin-right: 5px;
+          margin-left: 2px;
+        }
+
+        .prefix-reserve-label {
+          font-size: 14px;
+          font-weight: 400;
+          color: #2F6FA7;
```
Comment (zhhyu) [inline src/Bootstrap/less/theme/page-display-package-v2.less:61]: nit: nit: will it be better if we also use some shared color variable here? (For example: "color: @gray-light;")

**Review by zhhyu — COMMENTED** [review]
Nice work! :thumbsup:

**Pull request merged**